addons/sourcemod/scripting/ConsoleChatManager.sp

```diff
@@ -413,139 +444,30 @@ public Action SayConsole(int client, const char[] command, int args)
-		if(strcmp(soundp, "default") == 0)
-			FormatEx(soundt, 255, "common/talk.wav");
-		else
-			FormatEx(soundt, 255, soundp);
-	}
-
-	char sFinalText[1024];
-	char sCountryTag[3];
-	char sIP[26];
-	bool isCountable = IsCountable(sText);
-	bool containsDecimal = StringContainDecimal(sText);
-	bool isCountdown = !containsDecimal && isCountable;
-
-	for(int i = 1 ; i < MaxClients; i++)
-	{
-		if(IsClientInGame(i) && (!IsFakeClient(i) || IsClientSourceTV(i)))
-		{
-			if (g_bTranslation)
-			{
-				GetClientIP(i, sIP, sizeof(sIP));
-				GeoipCode2(sIP, sCountryTag);
-				KvGetString(kv, sCountryTag, sText, sizeof(sText), "LANGMISSING");
-
-				if (strcmp(sText, "LANGMISSING") == 0) KvGetString(kv, "default", sText, sizeof(sText));
-			}
-
-			FormatEx(sFinalText, sizeof(sFinalText), "%s", sText);
-
-			if (!g_bRemoveConsoleTag || g_bRemoveConsoleTag && (!ItContainSquarebracket(sText) || !ItContainColorcode(sText)))
-				FormatEx(sFinalText, sizeof(sFinalText), "%s%s", g_sConsoleTag, sText);
-
-			if(isCountable && GetRoundTimeAtTimerEnd() > 0)
-			{
-				float fMinutes = GetRoundTimeAtTimerEnd() / 60.0;
-				int minutes = RoundToFloor(fMinutes);
-				int seconds = GetRoundTimeAtTimerEnd() - minutes * 60;
-				char roundTimeText[32];
-
-				FormatEx(roundTimeText, sizeof(roundTimeText), " {orange}@ %i:%s%i", minutes, (seconds < 10 ? "0" : ""), seconds);
-				FormatEx(sFinalText, sizeof(sFinalText), "%s%s", sFinalText, roundTimeText);
-			}
-
-			CPrintToChat(i, sFinalText);
-
-			// Prepare HUD message
-			if (g_bEnableHud)
-			{
-				if (g_bHudMapSymbols)
-					RemoveDuplicatePrefixAndSuffix(sText);
+	SendServerMessage(sText, false);
 
-				RemoveTextInBraces(sText, true, true);
-
-				char szMessage[MAXLENGTH_INPUT + 10];
-				if (g_bHudSymbols)
-					FormatEx(szMessage, sizeof(szMessage), ">> %s <<", sText);
-				else
-					FormatEx(szMessage, sizeof(szMessage), "%s", sText);
-
-				PrepareHudMsg(i, szMessage, isCountdown);
-
-				if (isCountdown)
-					InitCountDown(szMessage);
-			}
-		}
-	}
-
-	if (g_bTranslation)
-	{
-		if(strcmp(soundp, "none", false) != 0)
-			EmitSoundToAll(soundt);
-
-		if(KvJumpToKey(kv, "hinttext"))
-		{
-			for(int i = 1 ; i < MaxClients; i++)
-				if(IsClientInGame(i) && (!IsFakeClient(i) || IsClientSourceTV(i)))
-				{
-					GetClientIP(i, sIP, sizeof(sIP));
-					GeoipCode2(sIP, sCountryTag);
-					KvGetString(kv, sCountryTag, sText, sizeof(sText), "LANGMISSING");
-
-					if (strcmp(sText, "LANGMISSING") == 0) KvGetString(kv, "default", sText, sizeof(sText));
-				
-					PrintHintText(i, sText);
-				}
-		}
-
-		KvRewind(kv);
-	}
 	return Plugin_Handled;
 }
 
+public MRESReturn Detour_ClientPrint(Handle hParams)
+{
+	// Check if message was sent from server console
+	int iPlayer = DHookGetParam(hParams, 1);
+	if (iPlayer != 0)
+		return MRES_Ignored;
+
+	// Check if the print was sent to chat
+	int iDestination = DHookGetParam(hParams, 2);
+	if (iDestination != 3)
```

Comment:
How do you know its chat? Could you use enum to clarify instead

Reply:
Added Enum struct, untested but it should work. cc @notkoen @maxime1907